boot/env.ml

```diff
@@ -54,21 +46,54 @@ let theories_dir = "theories"
 let plugins_dir = "plugins"
 let prelude = Filename.concat theories_dir "Init/Prelude.vo"
 
-let coqbin =
-  canonical_path_name (Filename.dirname Sys.executable_name)
+let find_in_PATH f =
+  match Sys.getenv_opt "PATH" with
+  | None -> None
+  | Some paths ->
+    let sep = if Coq_config.arch_is_win32 then ';' else ':' in
+    let paths = String.split_on_char sep paths in
+    paths |> List.find_opt (fun path ->
+        Sys.file_exists (if path = "" then f else Filename.concat path f))
+
+let rocqbin =
+  (* avoid following symlinks if possible (Sys.executable_name followed symlinks) *)
+  if Filename.basename Sys.argv.(0) <> Sys.argv.(0) then
```

Comment:
(about `cd user-contrib && dune exec`)
We could add a file_exists check here, but then find_in_PATH fails so we use executable_name which follows symlinks into the build layout, so we would need to understand the build layout.
Maybe in a followup PR but for now I'd rather get this merged.

Reply:
FWIW I now suspect this is actually a dune bug. I filed https://github.com/ocaml/dune/issues/11527. AFAICT `argv.(0)` is not *guaranteed* to actually contain a useful value but there's no reason to not pass a useful path if it exists.